### PR TITLE
cpu/esp32: define FLASHFILE_POS

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

Instead of using a fixed position of the image file in the flash, the variable `FLASHFILE_POS` is used which allows to override the default position of the image in the flash at 0x10000.

This PR is a prerequisite for the `periph_flashpage` implementation PR #19079.

### Testing procedure

Flashing a ESP32x SoC should work with `FLASHFILE_POS=0x20000`, for example:
```
USEMODULE=esp_log_startup FLASHFILE_POS=0x20000 BOARD=esp32-wroom-32 make -j8 -C tests/shell flash
```
The bootloader output should give `00020000` as offset for the `factory` partition
```
I (75) boot: Partition Table:
I (78) boot: ## Label            Usage          Type ST Offset   Length
I (84) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (91) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (97) boot:  2 factory          factory app      00 00 00020000 000199b0
I (104) boot: End of partition table
```
and
```
I (125) esp_image: segment 0: paddr=00020020 vaddr=3f400020 size=02140h (  8512) map
```
during the load of the image.

### Issues/PRs references

Prerequisite for PR #19079

### CODE REVIEW

File: cpu/esp32/Makefile.include [hunk 1]
```diff
@@ -151,6 +152,19 @@ else ifeq (120m,$(FLASH_FREQ))
   CFLAGS += -DCONFIG_ESPTOOLPY_FLASHFREQ_120M
 endif
 
+#extend CFLAGS by the corresponding FLASH_SIZE
+ifeq (1,$(FLASH_SIZE))
+  CFLAGS += -DCONFIG_ESPTOOLPY_FLASHSIZE_1MB
```
Comment (benpicco) [inline cpu/esp32/Makefile.include:157]: Would it make sense to use 
```suggestion
  CFLAGS += -DCONFIG_ESPTOOLPY_FLASHSIZE_$(FLASH_SIZE)MB
```
instead or is there only a fixed set of `CONFIG_` defines anyway?

Comment (gschorcht) [reply cpu/esp32/Makefile.include:157]: These are the values declared by ESP-IDF's Kconfig.

**Review by benpicco — APPROVED** [review]
bors merge

**Comment by gschorcht** [comment]
@benpicco Thanks for reviewing and merging.

**Pull request merged**